### ISSUE TITLE
Show why Retrieve/Deploy commands missing from menu

### ISSUE THREAD

**Issue by keirbowden** [forcedotcom/salesforcedx-vscode#764]
### Summary

I work across a number of projects concurrently and I'm using a combination of push/pull and retrieve/deploy functionality. I keep forgetting to set up everything correctly for the retrieve/deploy or forgetting that I'm using a scratch org, so often when I right click on a package.xml file I'm scratching my head as to why the menu options are missing. 

### Steps To Reproduce:

1.  Create a new sfdx project using the --manifest switch
1.  Create a package.xml file in the manifest directory
1. Right click and wonder why you can't deploy the code

### Expected result

It would be really nice if the context menu showed why the menu options weren't there - e.g. Deploy/retrieve not supported for scratch orgs, or Deploy/retrieve needs defaultusername. This would only be for the package.xml context menu, as by right clicking this it is clear that I'm expecting to be able to use the deploy/retrieve functionality.

### Actual result

No sfdx options in the context menu.

### Additional information

_Feel free to attach a screenshot_.

**VS Code Version**: VS Code 1.29 / Salesforce Extensions 44.7.0

**SFDX CLI Version**: SFDX CLI 6.40.0 / salesforcedx plugin 44.7.0

**OS and version**: MacOS 10.14

**Comment by ntotten** [comment]
I'm not really sure how we would implement this. I'm going to close it, but feel free to reopen if you have an idea of what the UI would look like.

**Comment by borkarameya** [comment]
What if the deploy/retrieve options were always shown? Whenever a user would try to deploy/retrieve the package.xml, the console would display an error. That way, the user isn't left wondering why they aren't able to deploy/retrieve.
It is really frustrating when we have authorized an org and the deploy/retrieve options don't show up in the context menu.

@ntotten Any thoughts on this?

**Comment by ntotten** [comment]
I certainly can see the frustration, but the problem with showing them all the time is that we need to load the extension on projects that aren't even Salesforce projects. This means that if you are developing a node.js project you would see these menus which I don't think is something people want.

The other option of showing them on all projects IMO would also cause frustration to some users. If somebody only develops with packages in scratch orgs why do they want to see menu items they cannot use?

I am open to proposals of how to implement this, but from what I can think of now, all options are bad options and I'm inclined to stick with what we have.

**Comment by nissimlevy** [comment]
I'm having the same problem. In visual studio code I used to see the deploy menu options when I right clicked on the default folder. Suddenly these options disappeared from the right click context menu. Why?

**Comment by nissimlevy** [comment]
No one wants to answer this.

**Comment by ntotten** [comment]
@nissimlevy Could you open a new issue with more details? There really isn't enough information to go on in your question to provide an answer. I am not sure this is the same issue as the original.

**Comment by nissimlevy** [comment]
I have installed Salesforce extensions and CLI. I have a project in Visual Studio Code version 1.32.3

I have authorized a Salesforce org and have deployed my code to this org successfully from Visual Studio Code using the right click context menu off the default folder. So I know I've set it up properly. 

The huge and frustrating problem is that most of the time when I open another project folder and then go back to the original folder I lose my context menu. I then have to create a new project and move my files to that project and then I have my context menu again until the next time that I lose it.
